lib/service/storage/elasticsearch.js

```diff
@@ -1387,6 +1387,20 @@ class ElasticSearch extends Service {
 
         await this.updateMapping(index, collection, mappings);
 
+        // update documents with new mappings
+        if (previousMappings.dynamic === 'false' && (await this._client.search(esRequest)).body.hits.hits.length > 0) {
```

Comment:
If think it's better to call `count` than `search` if you just want to know if there are documents to update

Side note:
Avoid putting long calls with chained property access like this in condition for clarity `await this._client.search(esRequest)).body.hits.hits.length` prefer storing the result in a const variable.
Conditions that are too long and uses `&&` or `||` should be splitted on multiple lines like so
```js
const documentsCount = await this._client.search(esRequest)).body.hits.hits.length;
if ( previousMappings.dynamic === 'false'
  && documentsCount > 0
) {
```